packages/blade/src/components/Checkbox/Checkbox.tsx

```diff
@@ -177,33 +177,34 @@ const Checkbox = ({
   });
 
   return (
-    <CheckboxLabel inputProps={state.isReactNative ? inputProps : {}}>
-      <CheckboxInput
-        isChecked={state.isChecked || Boolean(isIndeterminate)}
-        isDisabled={_isDisabled}
-        isNegative={_hasError}
-        inputProps={inputProps}
-      />
-      <CheckboxIcon
-        isChecked={state.isChecked}
-        isIndeterminate={isIndeterminate}
-        isDisabled={_isDisabled}
-        isNegative={_hasError}
-      />
-      <Box>
-        <CheckboxLabelText>{children}</CheckboxLabelText>
-        {showError && (
-          <FormHintText id={ids?.errorTextId} variant="error">
-            {errorText}
-          </FormHintText>
-        )}
-        {showHelpText && (
-          <FormHintText id={ids?.helpTextId} variant="help">
-            {helpText}
-          </FormHintText>
-        )}
-      </Box>
-    </CheckboxLabel>
+    <FormGroup
+      errorText={errorText}
+      errorTextId={ids?.errorTextId}
+      state={validationState === 'error' ? 'error' : 'help'}
+    >
+      <CheckboxLabel inputProps={state.isReactNative ? inputProps : {}}>
+        <CheckboxInput
+          isChecked={state.isChecked || Boolean(isIndeterminate)}
+          isDisabled={_isDisabled}
+          isNegative={_hasError}
+          inputProps={inputProps}
+        />
+        <CheckboxIcon
+          isChecked={state.isChecked}
+          isIndeterminate={isIndeterminate}
+          isDisabled={_isDisabled}
+          isNegative={_hasError}
+        />
+        <Box>
+          <CheckboxLabelText>{children}</CheckboxLabelText>
+          {showSupportingText && (
+            <SupportingText id={ids?.helpTextId} variant="help">
```

Comment:
What should be the scope of formgroup (example - supporting text is not in form group)? Also, aren't we planning to expose form group to consumers?

Reply:
SupportText is a very special usecase for Checkbox. 

I've extracted out helpText, errorText, successText logic into FormGroup so that it can be used in Input too. 

> Also, aren't we planning to expose form group to consumers?
Not right now, We have that as part of Milestone 3 in September. That time we will look to make the API more refined for public usage.